src/coreclr/tests/src/runtest.proj

```diff
@@ -385,15 +386,24 @@ namespace $([System.String]::Copy($(Category)).Replace(".","_").Replace("\","").
 
   <Target Name="GetListOfTestCmds">
     <ItemGroup>
-      <AllRunnableTestPaths Include="$(XunitTestBinBase)\**\*.cmd" Condition="'$(TargetOS)' == 'Windows_NT'" />
-      <AllRunnableTestPaths Include="$(XunitTestBinBase)\**\*.sh" Condition="'$(TargetOS)' != 'Windows_NT'" />
+      <AllRunnableTestPaths Include="$(XunitTestBinBase)\**\*.cmd" Condition="'$(TargetOS)' == 'Windows_NT' And '$(TargetArchitecture)' != 'wasm'" />
+      <AllRunnableTestPaths Include="$(XunitTestBinBase)\**\*.sh" Condition="'$(TargetOS)' != 'Windows_NT' And '$(TargetArchitecture)' != 'wasm'" />
+      <!-- On wasm, run-v8.sh is generated as part of running the tests; it should not be picked up as a test itself -->
+      <AllRunnableTestPaths Include="$(XunitTestBinBase)\**\*.sh" Exclude="$(XunitTestBinBase)\**\run-v8.sh" Condition="'$(TargetArchitecture)' == 'wasm'" />
```

Comment:
It may be a tad simpler to unconditionally collect all scripts as we used to and then conditionally <code>Remove</code> the run-v8.sh scripts from the item group for <code>wasm</code>

Reply:
I'm not sure I understand MSbuild well enough to know how to do that?

Reply:
I think it should be as follows (please try it out locally before pushing to the PR :-) ):

<pre>
&lt;AllRunnableTestPaths Remove="@(AllRunnableTestPaths)" Condition="'$(TargetArchitecture)' == 'wasm' && '%(Filename)%(Extension)' == 'run_v8.sh'" /&gt;
</pre>

Reply:
Or, alternatively,

<pre>
&lt;AllRunnableTestPaths Remove="$(XunitTestBinBase)\**\run-v8.sh" Condition="'$(TargetArchitecture)' == 'wasm'" /&gt;
</pre>

It's probably going to be a bit slower because it will scan the subtree twice but it may be preferable as it's easier to understand.